addons/binding/org.openhab.binding.bluetooth.bluez/src/main/java/org/openhab/binding/bluetooth/bluez/handler/BlueZBridgeHandler.java

```diff
@@ -254,6 +254,15 @@ private void checkForNewDevices() {
 
     private BlueZBluetoothDevice createAndRegisterBlueZDevice(tinyb.BluetoothDevice tinybDevice) {
         BlueZBluetoothDevice device = new BlueZBluetoothDevice(this, tinybDevice);
+        tinybDevice.getManufacturerData().keySet().stream().filter(manufacturerId -> manufacturerId != null).findFirst()
+                .ifPresent(manufacturerId -> {
+                    if (manufacturerId != null) {
```

Comment:
It looks like this check should not be needed?

Reply:
Oh right, nulls are filtered earlier. Will clean it up

Reply:
Now resolved in ca8d5e5